music21/spanner.py

```diff
@@ -471,6 +471,46 @@ def addSpannedElements(
 
         self.spannerStorage.coreElementsChanged()
 
+    def insertFirstSpannedElement(self, firstEl: base.Music21Object):
+        '''
+        Add a single element as the first in the spanner.
+
+        >>> n1 = note.Note('g')
+        >>> n2 = note.Note('f#')
+        >>> n3 = note.Note('e')
+        >>> n4 = note.Note('d-')
+        >>> n5 = note.Note('c')
+
+        >>> sl = spanner.Spanner()
+        >>> sl.addSpannedElements(n2, n3)
+        >>> sl.addSpannedElements([n4, n5])
+        >>> sl.insertFirstSpannedElement(n1)
+        >>> sl.getSpannedElementIds() == [id(n) for n in [n1, n2, n3, n4, n5]]
+        True
+        '''
+        from music21 import stream
+        elements: list[base.Music21Object] = self.getSpannedElements()
+        elOffsets: list[OffsetQL] = []
+        elActiveSites: list[stream.Stream] = []
+        for el in elements:
+            elOffsets.append(el.offset)
+            elActiveSites.append(el.activeSite)
+
+        # remove them all
+        for el in elements:
+            self.spannerStorage.remove(el)
+
+        # add firstEl first
+        self.addSpannedElements(firstEl)
+
+        # add all the rest in order
+        self.addSpannedElements(elements)
+
+        # restore all elements' activeSite and offset
+        for el, elOffset, elActiveSite in zip(elements, elOffsets, elActiveSites):
+            el.activeSite = elActiveSite
+            el.offset = elOffset
```

Comment:
please explain why all other elements need to be removed and then re-added?  It seems very inefficient, so I don't think we want to use this very much.

Reply:
It was the only way I could think of to insert at the first element.  Do you have a better way?

Reply:
I just tried simply doing self.spannerStorage.coreInsert(0, firstEl) followed by a coreElementsChanged call, but it still puts it at the end (which is correct, it is at offset 0, but I wanted it to be first).  @mscuthbert Is there some even more internal call I can make to insert at the start of spannerStorage.elements?

Reply:
This is still pending.